src/components/popup/details/purposes/purposes.jsx

```diff
@@ -65,18 +62,19 @@ export default class Purposes extends Component {
 	};
 
 	handleSelectSpecialFeatureOptins = ({isSelected, dataId}) => {
-		this.props.selectSpecialFeatureOptins(dataId, isSelected);
+		this.props.selectSpecialFeatureOptins(dataId+1, isSelected);
```

Comment:
In line 132 you send index+1 (which is good because you iterate over array) to this function. Hovewer inside this function you increment this value again. Thats why if you select special feature with id 1, special feature with id 2 is being set in model.

- [ ] wrong feature is being selected